src/item_location.cpp

```diff
@@ -22,7 +22,7 @@ class item_location::impl
     public:
         virtual ~impl() = default;
         virtual std::string describe( const Character * ) const = 0;
-        virtual int obtain( Character & ) = 0;
+        virtual int obtain( Character &ch, int qty ) = 0;
```

Comment:
`item::charges` is of type `long`, `qty` represents the same thing and should be a `long` as well. Note that `int` and `long` may not be the same / may not have the same size on all platforms.